librz/core/canalysis.c

```diff
@@ -7262,6 +7265,38 @@ RZ_IPI char *rz_core_analysis_all_vars_display(RzCore *core, RzAnalysisFunction
 	return rz_strbuf_drain(sb);
 }
 
+static int check_rom_exists(const void *value, const void *data) {
+	const char *name = (const char *)value;
+	const RzBinSection *sections = (const RzBinSection *)data;
+	return strcmp(name, sections->name);
+}
+
+RZ_API bool rz_analysis_add_io_registers_map(RzBinObject *o, RzAnalysis *analysis) {
```

Comment:
If you add a new API function - don't forget Doxygen please.

Reply:
Also please return `false` in case of either `o` and `analysis` are NULL. Also please add `rz_return_val_if_fail(o && analysis, false)`;.